### DISCUSSION TITLE
Cannot set/define CONFIG_UART_ASYNC_API to enable uart async api

### DISCUSSION THREAD

**Discussion by dj0405** [zephyrproject-rtos/zephyr#27728] [General] [answered]
I am trying to build the async uart test (zephyrproject/zephyr/tests/drivers/uart/uart_async_api) for stm32_min_dev_blue. I keep getting -ENOTSUP whenever uart_tx is called. 

The error comes from **include/drivers.uart.h:482**

```
static inline int z_impl_uart_tx(struct device *dev, const uint8_t *buf,
				 size_t len, int32_t timeout)

{
#ifdef CONFIG_UART_ASYNC_API
	const struct uart_driver_api *api =
			(const struct uart_driver_api *)dev->driver_api;

	return api->tx(dev, buf, len, timeout);
#else
	return -ENOTSUP;
#endif
}

```
CONFIG_UART_ASYNC_API is set to "y" in proj.conf, however, it does not appear in the build/zephyr/.confg and, therefore, the #ifdef test fails. I am not sure if this is a bug or I am missing something!

**proj.conf**

```
CONFIG_SERIAL=y
CONFIG_UART_ASYNC_API=y
CONFIG_ZTEST=y
CONFIG_TEST_USERSPACE=y
```

Appreciate your input/help.

**Comment by erwango** [comment, marked as answer]
@dj0405 UART_ASYNC_API is not yzet supported on STM32 serial driver.
You can check https://github.com/zephyrproject-rtos/zephyr/pull/14916 for a WIP implementation
